### PR TITLE
Base

### PR DESCRIPTION
**Pull request by DamianRivas**
Changes include...
* Upgrade Sprotty to latest
* Upgrade Theia to latest
* Add a search box to the diagram widget
* Implement structure for filtering elements in the diagram
* Create the `FilterAction` data structure to send filter requests to the backend server
* Override handlers for new actions
* Implement basic diagram editing
* Implement label editing

With the latest version of Theia, **Node.js v10** is needed to run it.

If changes are approved, I'll need to make changes to [this](https://github.com/DamianRivas/oml-theia/blob/f1ec74388c12a7c2fe43f2db8c3dfda9ac42cef0/.gitmodules#L3) line before merging.

### CODE REVIEW

File: io.opencaesar.oml.theia/travis-script.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
```
Comment (melaasar) [inline io.opencaesar.oml.theia/travis-script.sh]: Why is this new script needed?

Comment (DamianRivas) [reply io.opencaesar.oml.theia/travis-script.sh]: This script's purpose was to circumvent the console output limit on Travis CI. It's not needed anymore, but if we ever run into a situation where we need to run `yarn --verbose` or deal with any other verbose output, this script hides the output until the process fails and then dumps the last 1000 lines (adjustable) of the output to the console.

File: .travis.yml [hunk 2]
```diff
@@ -1,19 +1,25 @@
...
+node_js: '10'
 
 install:
+  - sudo apt install libx11-dev libxkbfile-dev libxkbfile-dev
   - curl -o- -L https://yarnpkg.com/install.sh | bash -s -- --version 1.12.3
   - export PATH=$HOME/.yarn/bin:$PATH ;
 
 before_script:
-  - cd io.opencaesar.oml.theia
+  - cd io.opencaesar.oml.theia && rm -rf oml-sprotty && git clone https://github.com/DamianRivas/oml-sprotty.git && cd oml-sprotty && git checkout base && cd ..
```
[the remark this answers is not in the document]
Comment (DamianRivas) [reply .travis.yml]: Removed. This was due to this PR being dependent on the sprotty PR as we discussed.

File: .travis.yml [hunk 3]
```diff
@@ -1,19 +1,25 @@
...
+  - secure: zs4oYaOFJZbU1VVNKplJwOsewU3sNujt9cYUzD1SpMrjqYQHEbrBXpX+WFET0nhWnCvb45i/Ui/vsUkAEPRFtn09ruF4mR+5pi5B5vt35fDqmK+vbXnpSuFGnHBQHV9TnvcLqaFkS+dlMV68DZXboxHt/X2Y8wFxVwvUKSQ0YBEdyiYWsQK3FM99c0bPzb26lG8Ps/iqOrOp/mfzd6YsN0oInqAAgTICLYIzq5w9MfjxGI9QEc6yxkwm6Mcgfq+BUut52ylDb+f5Sn12sr+m7BGZZ3J/NPL35KhR31KHuCMkr4fyxtL0mnfpstKRZYqYkBOj3rJefUZ1MtWWuCYfHMiBc9W/FOVaTA/25ls6cwUymwzqRa2nET3j01j0XQvPJgeuaX48Gz/nUVBmiqPlfZmtN5peFkZHPtYOa1rtcavtr7RvE900mhh1PubRrILJKGxmY4U/VGX3szRn7bc5jTHMsrzAChuFvvIcmRX3B+7D5258zl2fsGyhHvw3O1tSF0Y3JSP+BstyTfCH6sYPvwWYRCOIYxddyqvQirD7koN68Why9A4kZ2pLVMBxFjVksSfgAQ3cVz4c5Vbpo3fa0ciMecgMEObutOVtRKERWhh7xtxZeHPevFXBSAskEruPcKwIMshBfBlM2MVPAyPor141dzv2zrnLHqY0Nk+BzTk=
+  # - GITHUB_TOKEN=327e190c8069609e0311b70464ca3aab55f1bc0f
```
[the remark this answers is not in the document]
Comment (DamianRivas) [reply .travis.yml]: Removed

File: README.md [hunk 4]
```diff
@@ -9,16 +9,22 @@ An extension for the Theia IDE to support the [OML language server](https://gith
...
+```
 
 Install yarn.
 
+```
 npm install -g yarn
+```
 
 ## Clone
 ```
```
[the remark this answers is not in the document]
Comment (DamianRivas) [reply README.md]: Added. It will be there in my next push.

**Pull request merged**